src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,735 @@
+            Paths to the photons file, stars file, and log file.
+        """
+        suffix = self._label_suffix()
+        photons_file = self.photons_dir / f"incident_angles_photons_{suffix}.lis"
+        stars_file = self.photons_dir / f"incident_angles_stars_{suffix}.lis"
+        log_file = self.logs_dir / f"incident_angles_{suffix}.log"
+
+        if photons_file.exists():
+            try:
+                photons_file.unlink()
+            except OSError as err:
+                self.logger.error(f"Failed to remove existing photons file {photons_file}: {err}")
+
+        with photons_file.open("w", encoding="utf-8") as pf:
+            pf.write(f"#{'=' * 50}\n")
+            pf.write("# Imaging list for Incident Angle simulations\n")
+            pf.write(f"#{'=' * 50}\n")
+            pf.write(f"# config_file = {self.telescope_model.config_file_path}\n")
+            pf.write(f"# zenith_angle [deg] = {self.ZENITH_ANGLE_DEG}\n")
+            pf.write(
+                f"# off_axis_angle [deg] = {self.config_data['off_axis_angle'].to_value(u.deg)}\n"
+            )
+            pf.write(f"# source_distance [km] = {self.config_data['source_distance']}\n")
+
+        with stars_file.open("w", encoding="utf-8") as sf:
+            zen = self.ZENITH_ANGLE_DEG
+            dist = float(self.config_data["source_distance"])
+            sf.write(f"0. {90.0 - zen} 1.0 {dist}\n")
+
+        return photons_file, stars_file, log_file
+
+    def _write_run_script(self, photons_file, stars_file, log_file):
+        """Generate a run script for sim_telarray with the provided configuration and inputs.
+
+        Parameters
+        ----------
+        photons_file, stars_file, log_file : pathlib.Path
+            Input/output files for the run.
+
+        Returns
+        -------
+        pathlib.Path
+            Path to the generated shell script.
+        """
+        script_path = self.scripts_dir / f"run_incident_angles_{self._label_suffix()}.sh"
+        simtel_bin = self._simtel_path / "sim_telarray/bin/sim_telarray_debug_trace"
+        corsika_dummy = self._simtel_path / "sim_telarray/run9991.corsika.gz"
+
+        theta = self.ZENITH_ANGLE_DEG
+        off = float(self.config_data["off_axis_angle"].to_value(u.deg))
+        star_photons = self.config_data["number_of_photons"]
+
+        def cfg(par, val):
+            return f"-C {par}={val}"
+
+        opts = [
+            f"-c {self.telescope_model.config_file_path}",
+            f"-I{self.telescope_model.config_file_directory}",
+        ]
+        if self.perfect_mirror:
+            opts += [
+                "-DPERFECT_DISH=1",
+                "-C random_focal_length=0",
+                "-C mirror_reflection_random_angle=0",
+                "-C mirror_align_random_distance=0",
+                "-C mirror_align_random_horizontal=0,28,0,0",
+                "-C mirror_align_random_vertical=0,28,0,0",
+            ]
+
+        if self.mirror_reflection_random_angle is not None:
```

Comment:
Similar to early - does this overwrite parameters from the database when given from the command line?

Reply:
Yes, and I could remove this if not needed at all.